File: SoloLearn/It's a Sign/ItsaSign.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoloLearn
{
    class Program
    {

        static string saveSign() {
            string label, reverse;
            char[] charArray;
            for (int i=0; i<4; i++) {
                label = Console.ReadLine();
                charArray = label.ToCharArray();
                Array.Reverse(charArray);
                reverse = new string(charArray);
                if (label == reverse)
                    return "Open";
            }
            return "Trash";
        }

        static void Main(string[] args)
        {
            Console.WriteLine(saveSign());;
        }
    }
} 

File: SoloLearn/Code Coach/It's a Sign/ItsaSign.c
#include <stdio.h>
#include <string.h>

char* saveSign() {
    char label[50], reverse[50];
    int begin, end, count;
    for (int i = 0; i < 4; i++) {
        scanf("%s", label);
        strcpy(reverse, label);
        count = 0;
        while (label[count] != '\0')
            count++;
        end = count - 1;
        for (begin = 0; begin < count; begin++) {
            reverse[begin] = label[end];
            end--;
        } 
        reverse[begin] = '\0';
        if (strcmp(label, reverse) == 0)
            return "Open";
    }
    return "Trash";
}
int main() {
    printf("%s", saveSign());
    return 0;
} 